Makefile

```diff
@@ -37,7 +37,7 @@ GOIMPORTS=$(shell which goimports)
 
 .PHONY: fmt
 fmt: goimports
-	@for d in $$(go list -f {{.Dir}} ./...); do ${GOIMPORTS} --local github.com/hashicorp --local github.com/hashicorp/consul-api-gateway -w -l $$d/*.go; done
+	@for d in $$(go list -f {{.Dir}} ./...); do if [[ $$d != *"mocks"* ]]; then ${GOIMPORTS} --local github.com/hashicorp/consul-api-gateway,github.com/hashicorp -w -l $$d/*.go; fi; done
```

Comment:
so this is mostly working, we get a list of directories passed in which allows us to exclude anything in a `mocks` directory, excluding the other generated files like `pkg/apis/v1alpha1/zz_generated.deepcopy.go` is going to be a little more tricky, I was thinking of doing something hacky like running `git checkout $(git rev-parse --abbrev-ref HEAD) "*generated*"` after the fmt command completes, but that feels icky to me

Reply:
so we can actually get around this just diffing against fmt'd files in CI

Reply:
https://github.com/hashicorp/consul-api-gateway/pull/504/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR69

Reply:
🤔 Hmm, I think we might want to move ^ into the `make gen` task itself to avoid a false local diff if you forget to run `fmt` afterwards?

Reply:
oh good call, let me add that there as well

Reply:
and that actually should eliminate the need to run `fmt` in the ci pipeline because the command already handles it